src/listener/utils/transforms/flowdock.js

```diff
@@ -0,0 +1,142 @@
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+import { transform } from './resource-transformer';
+const url = require('url');
+
+function Title(resource) {
+  if (resource.type) {
+    if (resource.type.startsWith('AlertResource')) {
+      if (resource.alertTypeID && resource.alertTypeID.startsWith("Trap")) {
+        return "Trap: " + resource.description;
+      }
+      return resource.description + '\n';
+    }
+
+    if (resource.type.startsWith('server-hardware')) {
+      return 'Server Hardware: ' + resource.name + '\n';
+    }
+
+    if (resource.type.startsWith('ServerProfileTemplate')) {
+      return 'Profile Template: ' + resource.name + '\n';
+    }
+
+    if (resource.type.startsWith('ServerProfile') && !resource.type.startsWith('ServerProfileCompliancePreview')) {
+      return 'Profile: ' + resource.name + '\n';
+    }
+  }
+}
+
+function ToOutput(resource) {
+  const transformedRes = transform(resource);
+  let host;
+  if (resource.hyperlink) {
+    host = url.parse(resource.hyperlink).hostname;
+  }
+
+  const title = Title(transformedRes);
+
+  let output = '';
+  if (title) {
+    output = title;
+  }
+  if(transformedRes.pretext) {
+    output += transformedRes.pretext + '\n';
+  }
+  output += transformedRes.buildFlowdockOutput(host);
+  return output;
+}
+
+function output(resource) {
+  if (Array.isArray(resource)) {
+    return resource.map(ToOutput);
+  } else if (resource.members) {
+    return resource.members.map(ToOutput);
+  } else {
+    return [ToOutput(resource)];
+  }
+  return [];
+}
+
+export default class FlowdockTransform {
+  hyperlink(uri, name) {
+    return name ? name : uri;
+  }
+
+  list(lines) {
+    for (var i = 0; i < lines.length; i++) {
+      lines[i] = '  - ' + lines[i];
+    }
+    return lines;
+  }
+
+  text(msg, text) {
+    msg.send(text);
+  }
+
+  send(msg, resource, text) {
+    if (!resource) {
+      throw "Resource was null";
+    }
+
+    const out = output(resource).join('\n');
+    if (text) {
+      msg.send(out);
+    }
+
+    msg.send(out);
+  }
+
+  messageRoom(robot, room, resource, text) {
```

Comment:
does this need the same logic you used in: oneview.js

Reply:
I'll fix it when I make refactoring.